### PR TITLE
Ensure that KC remembers the role you last used for a particular account

### PR DESCRIPTION
**Pull request by trinitroglycerin**

### CODE REVIEW

File: cli/config.go
Comment (trinitroglycerin) [inline cli/config.go:147]: I'm not sure what I was thinking with these 4 lines, it was very stupid. This would also have broken attempts at parsing configs with old clients too.

This is why you should unit test everything

File: cli/config.go
Comment (trinitroglycerin) [inline cli/config.go:269]: We put this back to a pointer so we can modify the role name easily

File: cli/get.go
Comment (riot-jetaylor) [inline cli/get.go:83]: do we prefer a most recent role over a user configured default? I can see a case where you may cycle through roles and want the tool provided role to not be the last one.

Comment (trinitroglycerin) [reply cli/get.go:83]: The ask was specifically the most recent role from users

Comment (riot-jetaylor) [reply cli/get.go:83]: for now, this seems to work for our users

**Review by riot-jetaylor — DISMISSED** [review]

**Comment by trinitroglycerin** [comment]
I didn't dismiss any reviews, I just mistakenly pushed a new commit to my main branch before this one was merged. I'm merging this as we already did a review of this PR and it has not changed

**Pull request merged**
